### PR TITLE
Allow GltfLoader to load glTF from a parsed JSON

### PR DESCRIPTION
**Pull request by ptrgags**
Don't merge this until #9725 is merged. I branched off there to avoid some merge conflicts.

When a tileset with glTF (JSON not binary) files is loaded, the JSON is parsed automatically (because this is the only way to distinguish external tilesets from glTF models). However, this means `Gltf3DTileContent` recieves the parsed object, not a typed array.

This PR changes GltfLoader to allow a gltfJson directly to fix this.

[Local Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=bVA9T8MwEP0rlqcgIVuIDdIIKaUsIIZGTF6c5FoMFzuyLykt4r9jN5VoS+3Bunsf984lBDN04vGrB286sKRxAZoGD0GA1TXCi2sBj3E2Y+QHuFd21J6NBjbgY8/ChpWT29u+lyne7OvSRZmx4BW/iqp0k5IMQgA6lU7P7byawOxbWcYGj3dMcbnsoQlyrknLE558QlqlKTGfXGO1kAdv8RGcVVzZn2nwFFaEBiyIPi5kyIxxU9222UGSiAfazrmucn8Av+Z5oC1CkUKl82C63nlKATMhJEHXo6YYqB6azzi+CSH5JWouj6V5a0Zm2tmFL2IN6hAishoQl2YHihe5jPx/UnS6NXb9OoJHvU2095vieWoKIXIZy8tKcg5r7c+cfwE)

@sanjeetsuhag could you review? this is complete, just don't merge yet

### CODE REVIEW

File: Source/Scene/GltfLoader.js [hunk 1]
```diff
@@ -188,6 +190,13 @@ Object.defineProperties(GltfLoader.prototype, {
  * @private
  */
 GltfLoader.prototype.load = function () {
+  // If we have a parsed Gltf JSON, we can short circut
```
Comment (sanjeetsuhag) [inline Source/Scene/GltfLoader.js:193]:
```suggestion
  // Skip loading glTF JSON if it was passed in as an argument.
```

**Review by sanjeetsuhag — COMMENTED** [review]
@ptrgags Code looks good, just a minor doc fix needed.

**Comment by ptrgags** [comment]
@sanjeetsuhag updated!

**Comment by ptrgags** [comment]
Undrafted this PR now that #9725 is merged and synced with `main` to pull in #9770 to avoid Travis woes

@lilleyse could you do a final review?

File: Source/Scene/ModelExperimental/ModelExperimental.js [hunk 2]
```diff
@@ -380,8 +380,8 @@ ModelExperimental.prototype.destroy = function () {
  * The model can be a traditional glTF asset with a .gltf extension or a Binary glTF using the .glb extension.
  *
  * @param {Object} options Object with the following properties:
- * @param {String|Resource|Uint8Array} [options.gltf] A Resource/URL to a glTF/glb file or a binary glTF buffer
- * @param {Object} [options.basePath=''] The base path that paths in the glTF JSON are relative to.
+ * @param {String|Resource|Uint8Array|Object} [options.gltf] A Resource/URL to a glTF/glb file, a binary glTF buffer, or a JSON object containing the glTF contents
```
Comment (lilleyse) [inline Source/Scene/ModelExperimental/ModelExperimental.js:383]: The property is not optional. Must have overlooked that from a previous review.

```suggestion
 * @param {String|Resource|Uint8Array|Object} options.gltf A Resource/URL to a glTF/glb file, a binary glTF buffer, or a JSON object containing the glTF contents
```

File: Source/Scene/GltfLoader.js [hunk 3]
```diff
@@ -188,6 +190,13 @@ Object.defineProperties(GltfLoader.prototype, {
  * @private
  */
 GltfLoader.prototype.load = function () {
+  // Skip loading glTF JSON if it was passed in as an argument.
+  if (defined(this._gltfJson)) {
+    this._state = GltfLoaderState.LOADED;
+    this._textureState = GltfLoaderState.LOADED;
+    return;
+  }
+
```
Comment (lilleyse) [inline Source/Scene/GltfLoader.js:199]: `GltfJsonLoader` can't be skipped, there's a lot of work it does besides fetching the glTF, like loading external resources and decoding data uris.

Comment (lilleyse) [reply Source/Scene/GltfLoader.js:199]: The JSON will need to be passed through similar to how `typedArray` is passed through. Hopefully pretty straightforward using `typedArray` as a reference.

**Comment by lilleyse** [comment]
@ptrgags your updates look good, just add a test to `GltfJsonLoaderSpec.js`

**Comment by ptrgags** [comment]
Updated! Though since @lilleyse has a packed schedule today, @ebogo1 could you do the final review?

**Comment by lilleyse** [comment]
I revised the test. It seems like it was somewhat hastily modified from the "loads typed array" spec including the variable still being called `gltf2Binary` despite it being a json object. I made it closer to the "loads glTF 2.0" test instead.

I'll merge when CI passes.

**Pull request merged**